public/js/length_visualisation.js

```diff
@@ -0,0 +1,247 @@
+import _ from 'underscore';
+import * as Helpers from './visualisation_helpers';
+
+export default class LengthVisualisation {
+  constructor(query, $svg_container, algorithm) {
+    this._svg_container_d3 =  d3.select($svg_container[0]);
+    this.query = query;
+    this._seq_type = Helpers.get_seq_type(algorithm);
+
+    this._margin = {top: 30, right: 60, bottom: 70, left: 25};
+    this._width = $svg_container.width() - this._margin.left - this._margin.right;
```

Comment:
`width` and `height` are used only inside the constructor. They thus don't need to be a property.